### PR TITLE
OCPBUGS-28647: deferred updates: cleanups

### PR DESCRIPTION
**Pull request by ffromani**
cleanups suggested late in the #1019 cycle or emerged after the first nightly CI run

### CODE REVIEW

**Comment by jmencak** [comment]
Thank you for the cleanup.  This looks good to me, but letting other reviewers comment.  Only one super-nit:
```
diff --git a/pkg/tuned/controller.go b/pkg/tuned/controller.go
index cce3334a..561a60dc 100644
--- a/pkg/tuned/controller.go
+++ b/pkg/tuned/controller.go
@@ -403,7 +403,7 @@ type ExtractedProfiles struct {
 
 // ProfilesExtract extracts TuneD daemon profiles to tunedProfilesDirCustom directory.
 // Returns:
-//   - ExtractedProfile with the details of the operation performed
+//   - ExtractedProfiles with the details of the operation performed.
 //   - Error if any or nil.
 func ProfilesExtract(profiles []tunedv1.TunedProfile, recommendedProfile string) (ExtractedProfiles, error) {
 	klog.Infof("profilesExtract(): extracting %d TuneD profiles (recommended=%s)", len(profiles), recommendedProfile)

```

**Comment by ffromani** [comment]
> Thank you for the cleanup. This looks good to me, but letting other reviewers comment. Only one super-nit:
> 
> ```
> diff --git a/pkg/tuned/controller.go b/pkg/tuned/controller.go
> index cce3334a..561a60dc 100644
> --- a/pkg/tuned/controller.go
> +++ b/pkg/tuned/controller.go
> @@ -403,7 +403,7 @@ type ExtractedProfiles struct {
>  
>  // ProfilesExtract extracts TuneD daemon profiles to tunedProfilesDirCustom directory.
>  // Returns:
> -//   - ExtractedProfile with the details of the operation performed
> +//   - ExtractedProfiles with the details of the operation performed.
>  //   - Error if any or nil.
>  func ProfilesExtract(profiles []tunedv1.TunedProfile, recommendedProfile string) (ExtractedProfiles, error) {
>  	klog.Infof("profilesExtract(): extracting %d TuneD profiles (recommended=%s)", len(profiles), recommendedProfile)
> ```

thanks, fixed

**Comment by ffromani** [comment]
/cc @Tal-or 
as he did most of the comments I'm addressing here

File: test/e2e/deferred/restart.go [hunk 1]
```diff
@@ -20,6 +22,23 @@ import (
...
+	}
+	val, err := strconv.ParseBool(envVal)
+	if err != nil {
+		util.Logf("wrong value for %q: %q (err=%v)", enableRestartTestsEnvVar, envVal, err)
+		return false
+	}
+	return val
+}
+
 var _ = ginkgo.Describe("[deferred][restart] Profile deferred", func() {
```
Comment (Tal-or) [inline test/e2e/deferred/restart.go:42]: Maybe we should also add `[disruptive]` ? 
https://github.com/kubernetes/community/blob/0566459c2bdfec36670861be20be26d33f31467e/contributors/devel/sig-testing/e2e-tests.md#kinds-of-tests

File: test/e2e/deferred/restart.go [hunk 2]
```diff
@@ -20,6 +22,23 @@ import (
 	"github.com/openshift/cluster-node-tuning-operator/test/e2e/util/wait"
 )
 
+const (
+	enableRestartTestsEnvVar = "DEFERRED_RESTART_TESTS"
+)
+
+func areRestartTestsEnabled() bool {
```
Comment (Tal-or) [inline test/e2e/deferred/restart.go:29]: `Ginkgo` has a built-in Labels mechanism that allow you to skip/unskip tests in a more natural manner. 
Unless the commands the executes the tests sits somewhere else (on different code-base) then we need to actively configure it to skip the label, which is less convenient.

File: test/e2e/deferred/restart.go (hunk 1 above)
Comment (ffromani) [reply test/e2e/deferred/restart.go:42]: indeed, done

File: test/e2e/deferred/restart.go (hunk 2 above)
Comment (ffromani) [reply test/e2e/deferred/restart.go:29]: very good point, let's add labels, and let's fix the matching tags

**Comment by Tal-or** [comment]
@ffromani I don't see, where the actual skip is done?
We should have something like this:
https://github.com/openshift/cluster-node-tuning-operator/blob/master/Makefile#L260C135-L260C162

**Comment by ffromani** [comment]
> @ffromani I don't see where the actual skip is done? We should have something like this: https://github.com/openshift/cluster-node-tuning-operator/blob/master/Makefile#L260C135-L260C162 Am I missing something here?

because is not yet :) I'll still need to do the skipping part consuming the labels

**Comment by ffromani** [comment]
/hold

incomplete PR

**Comment by ffromani** [comment]
/hold cancel

xref: https://github.com/openshift/release/blob/master/ci-operator/config/openshift/cluster-node-tuning-operator/openshift-cluster-node-tuning-operator-master.yaml#L119

**Comment by Tal-or** [comment]
/lgtm
Thanks!

**Comment by ffromani** [comment]
/unhold

before the latest update:
```
$ make test-e2e
for d in deferred; do \
  KUBERNETES_CONFIG="/home/fromani/clusters/cnflab/kubeconfig" GOOS=linux GO111MODULE=on GOFLAGS=-mod=vendor go test -v -timeout 40m ./test/e2e/$d -ginkgo.v -ginkgo.no-color -ginkgo.fail-fast -ginkgo.label-filter=\"!flaky\" || exit; \
done
=== RUN   TestNodeTuningOperatorDeferred
Ginkgo detected configuration issues:
Syntax Error Parsing Label Filter
  "!flaky"
  Invalid token '!'.

  Learn more at: http://onsi.github.io/ginkgo/#spec-labels
FAIL	github.com/openshift/cluster-node-tuning-operator/test/e2e/deferred	0.021s
FAIL
make: *** [Makefile:103: test-e2e] Error 1
```

**Comment by Tal-or** [comment]
The label is `Flaky` (begin with capital letter) but the filter is `!flaky` (small letter). 
Are you sure it would match the label?

**Comment by ffromani** [comment]
> The label is `Flaky` (begin with capital letter) but the filter is `!flaky` (small letter). Are you sure it would match the label?

good catch. Partial update. Fixing.

**Comment by ffromani** [comment]
uhm, kube standard is to use `Capitalized` nouns, but the rest of the tests is using `lowercase-dash-separated`

**Comment by Tal-or** [comment]
/lgtm
Thank you for the fixes

**Comment by ffromani** [comment]
thank you for the careful review! low caffeine intake today

**Comment by jmencak** [comment]
> New changes are detected. LGTM label has been removed.

Thank you @ffromani , the last change was requested by me.

/lgtm

**Pull request merged**